addons/portal/static/src/js/portal.js

```diff
@@ -46,6 +49,41 @@ publicWidget.registry.portalDetails = publicWidget.Widget.extend({
     _onCountryChange: function () {
         this._adaptAddressForm();
     },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onEditProfilePicClick: function (ev) {
+        ev.preventDefault();
+        ev.currentTarget.closest("form").querySelector(".o_portal_file_upload").click();
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onFileUploadChange: function (ev) {
+        if (!ev.currentTarget.files.length) {
+            return;
+        }
+        const formEl = ev.currentTarget.closest("form");
+        var reader = new window.FileReader();
+        reader.readAsDataURL(ev.currentTarget.files[0]);
+        reader.onload = function (ev) {
+            formEl.querySelector(".o_wportal_avatar_img").src = ev.target.result;
+        };
+        formEl.querySelector("#portal_clear_image")?.remove();
+    },
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onProfilePicClearClick: function (ev) {
+        const formEl = ev.currentTarget.closest("form");
+        formEl.querySelector(".o_wportal_avatar_img").src = "/web/static/img/placeholder.png";
```

Comment:
Hello @panv-odoo, we need an additional action here. After deleting the profile picture and replacing it with a placeholder, we also need the delete button to be removed. 

Could you add toggle functionality so that when an image is uploaded, the delete icon reappears? Currently, after deleting the image, it looks like this:
![image](https://github.com/user-attachments/assets/ff933097-fa6b-42de-b2ab-1538dfb34648)
and the delete button is left without a purpose.